src/components/Character.hpp

```diff
@@ -56,7 +86,38 @@ namespace REGoth
     void createInventoryItem(const bs::String& instance, bs::INT32 num);
 
     void setToFightMode(const bs::String& weapon);
-    void setToFistMode();
+
+    /**
+     * @return The waypoint this character is around, needed by the scripts.
+     */
+    const bs::String& currentWaypoint() const;
+
+    /**
+     * Sets the waypoint this character is around, needed by the scripts.
+     * This could be the waypoint of the active routine, for example.
+     */
+    void setCurrentWaypoint(const bs::String& waypoint);
+
+    /**
+     * @return On monsters, this will return the AI-state the monster should
+     *         automatically start after spawning.
+     *         On Humans, this will return an empty string.
+     */
+    const bs::String& getStartAIState();
+
+    /**
+     * @return Name of the script function to call for the daily routine.
+     */
+    bs::String dailyRoutine();
+    void setDailyRoutine(const bs::String& newDailyRoutine);
```

Comment:
Could you add a small comment to this function? In particular what `newDailyRoutine` might be.

Reply:
Of course. Most of these methods are just stubs and I think many of them would be better off in the AI component. I just dumped everything gothic had in it's original engine which was a script external and started with NPC_ here.

I think it's a flaw in the original design which they had to make because they had the event queue also inside the npc class. We have a separate component for it and can clean this up a bit.